src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULT_ARRAYLIST = {};
+    private Object[] objectArray;
+
+    private int size;
+
+    public ArrayList() {
+        objectArray = DEFAULT_ARRAYLIST;
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.objectArray = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.objectArray = DEFAULT_ARRAYLIST;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (size < index || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Cen`t add element. "
+                    + "Index " + index + " is out of range 0 - " + size);
+        }
+        if (objectArray.length == 0) {
+            objectArray = new Object[DEFAULT_CAPACITY];
+        }
 
+        if (objectArray.length < size + 1) {
+            objectArray = Arrays.copyOf(objectArray, objectArray.length
+                    + (objectArray.length >> 1));
+        }
+        System.arraycopy(objectArray, index, objectArray, index + 1,
+                (objectArray.length - 1) - index);
+        objectArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T get(int index) {
-        return null;
+        if (index > (size - 1) || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Cen`t get element. "
+                    + "Index " + index + " is out of range 0 - " + (size - 1));
+        }
```

Comment:
It's better to create private method for this check. You use this check multiple times. Use it in other places
```suggestion
        checkIndex(index);
```